ImmersiveEquipmentMeshGen/MeshHandler.cs

```diff
@@ -337,8 +337,33 @@ internal void TransformMeshes()
             IDictionary<string, string> bsaDone = new ConcurrentDictionary<string, string>();
             if (bsaFiles.Count > 0)
             {
+                List<Noggog.FilePath> archivePaths = new();
+
+                // create archive path list according to load order
+                foreach (var e in ScriptLess.PatcherState.LoadOrder)
+                {
+                    foreach (var f in Archive.GetApplicableArchivePaths(GameRelease.SkyrimSE, ScriptLess.PatcherState.DataFolderPath, e.Value.ModKey).Reverse())
+                    {
+                        // GetApplicableArchivePaths includes base archives for each mod
+                        if (!archivePaths.Contains(f))
+                        {
+                            archivePaths.Insert(0, f);
+                        }
+                    }
+                }
```

Comment:
I think this can be simplified a bit via Linq:
```
var archivePaths = ScriptLess.PatcherState.LoadOrder.PriorityOrder
    .SelectMany(x => Archive.GetApplicableArchivePaths(GameRelease.SkyrimSE, ScriptLess.PatcherState.DataFolderPath))
    .Distinct()
    .ToArray();
```

Uses the `PriorityOrder` to clarify the order we want things to be on the list, and uses `Distinct` to handle the `Contains` logic (which will be faster, as it'll use a hashset under the hood instead of a List<string> of paths, which is generally very slow, relatively)

Reply:
Additionally, I think we could consider adding a new Mutagen API call taking in load order:
```
var archivePaths = Archive.GetApplicableArchivePaths(GameRelease.SkyrimSE, ScriptLess.PatcherState.DataFolderPath, ScriptLess.PatcherState.LoadOrder.PriorityOrder)
    .Distinct()
    .ToArray();
```

At first glance, it might seem like we'd just do the loop/linq from above internally, but I think it can do a bit more than that.  bsas included by being listed in the ini throw some curveballs, which a more global "give me the environment bsas" could return more naturally.

End goal i think could be exposing Archives in the GameEnvironment/SynthesisState systems.  Eg, just like `PatcherState` has a `LoadOrder`, it could also expose `PatcherState.Archives` with some object that was loopable for normal iteration, and then has some accessors to get specific archives as needed.  This would replace the more static access of `Archive.SomeStaticCall`

TLDR:
I think the current code is fine.  We can likely refactor and revisit later once I upgrade stuff

Reply:
```
var archivePaths = ScriptLess.PatcherState.LoadOrder.PriorityOrder
    .SelectMany(x => Archive.GetApplicableArchivePaths(GameRelease.SkyrimSE, ScriptLess.PatcherState.DataFolderPath))
    .Distinct()
    .ToArray();
```
Since we're looking make an archive priority order for the loop below, we need: 

1.) A _single_ listing from sResourceArchiveList(2) with the order preserved
2.) Applicable archives from each mod in the LO in the correct order (I think '* - Textures.bsa' take priority, need to check)

ATM this needs to be carefully constructed since `GetApplicableArchivePaths` returns sResourceArchiveList even when `ModKey` argument is passed. 

Simplified it would look something like this:

```
var archivePaths = ScriptLess.PatcherState.LoadOrder.ListedOrder
  .SelectMany(x => Archive.GetApplicableArchivePaths(GameRelease.SkyrimSE, ScriptLess.PatcherState.DataFolderPath, x.ModKey).Reverse())
  .Distinct()
  .Reverse()
```

Example output:

```
RaceMenu.bsa
SkyUI.bsa
Wyrmstooth - Textures.bsa
Wyrmstooth.bsa
LegacyoftheDragonborn - Textures.bsa
LegacyoftheDragonborn.bsa
3DNPC.bsa
BS_DLC_patch.bsa
BSHeartland - Textures.bsa
BSHeartland.bsa
BSAssets - Textures.bsa
BSAssets.bsa
Unofficial Skyrim Special Edition Patch - Textures.bsa
Unofficial Skyrim Special Edition Patch.bsa
Skyrim - Patch.bsa
Skyrim - Textures8.bsa
Skyrim - Textures7.bsa
Skyrim - Textures6.bsa
Skyrim - Textures5.bsa
Skyrim - Textures4.bsa
Skyrim - Textures3.bsa
Skyrim - Textures2.bsa
Skyrim - Textures1.bsa
Skyrim - Textures0.bsa
Skyrim - Voices_en0.bsa
Skyrim - Sounds.bsa
Skyrim - Meshes1.bsa
Skyrim - Meshes0.bsa
Skyrim - Animations.bsa
Skyrim - Interface.bsa
Skyrim - Shaders.bsa
Skyrim - Misc.bsa
```

`PatcherState.Archives` with equivalent `ListedOrder`/`PriorityOrder` sounds great!

Reply:
Yeah, i do think this Archive exposure system needs a revamp.  Thanks for the detail, and ill be taking a closer look when I get to focusing on upgrades